### PR TITLE
Use eslint

### PR DESCRIPTION
**Pull request by Tyrrrz**
This adds ESLint to catch potential errors in our `.vue` and `.js` files.

Also combines all checks into one task.

### CODE REVIEW

**Comment by Tyrrrz** [comment]
Need to merge #51 first

**Comment by Tyrrrz** [comment]
Something is broken, I'll figure this one out

**Comment by Tyrrrz** [comment]
Ok @abergs I'm stuck a bit on this one:

image

Where does `Passwordless` come from in this context? It's not imported anywhere.

https://github.com/passwordless/docs/blob/048add2c81a100bd68a27c7ca2ef5ddfff9b3dab/src/.vuepress/components/PasswordlessDemo.vue#L34

**Comment by Tyrrrz** [comment]
I decided to go with approach 2

**Comment by Tyrrrz** [comment]
Looks like Cloudflare Pages is failing but I don't have access to the logs to know why

File: src/.vuepress/components/PasswordlessDemo.vue [hunk 1]
```diff
@@ -53,6 +40,12 @@ export default {
       platform: false
     };
   },
+  mounted() {
+    const cached = localStorage.getItem('isPlatformSupporterd');
```
Comment (jonashendrickx) [inline src/.vuepress/components/PasswordlessDemo.vue:44]: Looks like a typo
```suggestion
    const cached = localStorage.getItem('isPlatformSupported');
```

**Review by jonashendrickx — CHANGES_REQUESTED** [review]

**Comment by abergs** [comment]
Seems you figured it out - LGTM.

Sidenote: we should add that demo button to the docs where it makes sense.

**Review by jonashendrickx — APPROVED** [review]

**Pull request merged**